packages/web-app-files/src/mixins/fileActions.js

```diff
@@ -75,11 +75,15 @@ export default {
           handler: item =>
             this.$_fileActions_openEditor(editor, item.path, item.id, EDITOR_MODE_EDIT),
           isEnabled: ({ resource }) => {
-            if (editor.routes?.length > 0 && !checkRoute(editor.routes, this.$route.name)) {
-              return false
+            if (
+              editor.handler ||
```

Comment:
The idea of requiring the definition of valid routes for a file editor was, that we have a way to hide something (mainly for the trashbin). With this change it would be impossible to block e.g. your OnlyOffice handler on the trashbin route. The route check is not good, because it requires your extension to have domain knowledge about the routing of the files app. But it gets the job done for now, so I'd like to keep it like it is now. What do you think?

Reply:
Ahh okay - sounds like trashbin is a special case but I get where you are coming from, happy to leave this check as is, but would like the `fileId` change to stay, are you happy with that?  If so i'll update the PR and changelog appropriately.

Reply:
Cool, thanks for understanding :-) The `fileId` changes are very welcome, would be happy to merge your PR once it's updated. 👍 Thanks for taking care!

Reply:
Will do cheers.

Reply:
@rpocklin may I ping you and ask to rebase & finalize this one, or should we take over? 😇

Reply:
Hi @pascalwengerter apologies please if you can take this over i'd appreciate it.